src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,217 @@
+import React from 'react';
+import styled, { ThemeProps, css } from 'styled-components';
+import { variant } from 'styled-system';
+import { spacing, pxToRem, typography, Size } from 'shared';
+import { MdCheckBox } from 'react-icons/md';
+import { MdOutlineCheckBoxOutlineBlank } from 'react-icons/md';
+import { MdIndeterminateCheckBox } from 'react-icons/md';
+import { Label } from 'components/Label';
+import { HelperText } from 'components/HelperText';
+
+/**
+ * Various dimensions of checkbox component
+ */
+const checkboxDimensions = {
+  contentSpacing: spacing(1),
+  checkboxMargin: pxToRem(21.1),
+  helperTextFontSize: pxToRem(14),
+  small: {
+    fontSize: pxToRem(16),
+  },
+  large: {
+    fontSize: pxToRem(24),
+  },
+};
+
+/**
+ * Internal properties for styles
+ */
+interface InnerProps {
+  /**
+   * Disable checkbox
+   */
+  disabled?: boolean;
+
+  /**
+   * Checkbox is indeterminate state
+   */
+  indeterminate?: boolean;
+
+  /**
+   * Size of checkbox
+   */
+  size?: Exclude<Size, 'medium'>;
+
+  /**
+   * Checkbox helper text
+   */
+  helperText?: string;
+
+  /**
+   * Change event handler passed from internal component
+   */
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+/**
+ * Checkbox component properties
+ * Extrends html input element attributes
+ * https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#additional_attributes
+ */
+export interface CheckboxProps
+  extends InnerProps,
+    Omit<React.InputHTMLAttributes<HTMLInputElement>, 'size'> {
+  /**
+   * Checkbox label text
+   */
+  label?: string;
+
+  /**
+   * Checkbox is checked
+   */
+  checked?: boolean;
+}
+
+/**
+ * Helper function to calculate correct sizes for font size
+ * @param props mandatory checkbox props
+ * @returns modified css
+ */
+const calculateSizes = (props: InnerProps) => {
+  return css`
+    ${variant({
+      prop: 'size',
+      variants: {
+        small: {
+          fontSize: checkboxDimensions.small.fontSize,
+        },
+        large: {
+          fontSize: checkboxDimensions.large.fontSize,
+        },
+      },
+    })};
+  `;
+};
+/**
+ * Helper function for checkbox disabled styles
+ * @param props mandatory checkbox props and theme props
+ * @returns modified css
+ */
+const isDisabled = (props: ThemeProps<any> & InnerProps) => {
+  return css`
+    ${variant({
+      prop: 'disabled',
+      variants: {
+        true: {
+          color: props.theme.colors.digitalBlack300,
+          cursor: 'default',
+        },
+        false: {
+          color: props.theme.colors.digitalBlack,
+          cursor: 'pointer',
+        },
+      },
+    })};
+  `;
+};
+
+/**
+ * calculate checkbox placement margin based on helper text
+ * @param props mandatory checkbox props
+ * @returns modified css
+ */
+const calculateMargin = (props: InnerProps) => {
+  return css`
+    margin-bottom: ${() =>
```

Comment:
En nyt ihan satavarma ole kokeilematta mutta mun mielestä ternary tässä pitäis onnistua myös ilman arrow-funktiota

Reply:
poistettu arrow funkkari